gsuite/resource_user.go

```diff
@@ -621,7 +643,16 @@ func resourceUserRead(d *schema.ResourceData, meta interface{}) error {
 func resourceUserDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	err := config.directory.Users.Delete(d.Id()).Do()
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		err := config.directory.Users.Delete(d.Id()).Do()
+		if err == nil {
+			return nil
+		}
+		if gerr, ok := err.(*googleapi.Error); ok && (gerr.Errors[0].Reason == "quotaExceeded" || gerr.Code == 429) {
```

Comment:
should the `gerr.Errors[0].Reason == "quotaExceeded"` condition just be included in the retryTime function in utils?

Reply:
It is; when I grabbed this code from the google provider, it looks like they allow less failure scenarios for deletes, hence why I copied that functionality here too, I'm not sure if it matters for the directory API though.

Reply:
I may be wrong, but isn't 429 is what you would get when you exceed your API quota, making the test redundant?

Reply:
Nope:
```
* gsuite_group.test[68]: gsuite_group.test.68: googleapi: Error 403: Request rate higher than configured., quotaExceeded
```

Reply:
I see, my bad, I'm not familiar with that API.